src/analysis/normed/group/seminorm.lean

```diff
@@ -51,44 +51,88 @@ structure group_seminorm (G : Type*) [group G] :=
 
 attribute [nolint doc_blame] add_group_seminorm.to_zero_hom
 
+/-- `add_group_seminorm_class F α` states that `F` is a type of seminorms on the additive group `α`.
+
+You should extend this class when you extend `add_group_seminorm`. -/
+class add_group_seminorm_class (F : Type*) (α : out_param $ Type*) [add_group α]
```

Comment:
Could you also define a class for `absolute_value`, since it shares some fields with this one?

Reply:
Do you mean their common ancestor, then? Seminorms are not an ancestor of absolute values because they are real-valued and absolute values are not an ancestor of seminorms because they are multiplicative.

Another solution would be to generalize seminorms away from `ℝ` but that should definitely be a separate PR.

Reply:
Yes exactly, I think we should have a `map_add_le_class` (or whatever) that both `absolute_value` and `add_group_seminorm_class` inherit from.

Reply:
Okay, I've added the barebones ones. I could add a few more for `f x = 0 → x = 0` and intersection of such, so that we can generalize the remaining `group_seminorm` lemmas, but I don't know if that's worth it.

Reply:
Like this is fine for now, thanks!